tests/components/apprise/test_notify.py

```diff
@@ -0,0 +1,68 @@
+"""The tests for the apprise notification platform."""
+from unittest.mock import patch
+
+from homeassistant.setup import async_setup_component
+
+BASE_COMPONENT = "notify"
+
+
+async def test_apprise_config_load_fail(hass):
+    """Test apprise configuration failures."""
+
+    config = {"apprise": {"name": "test", "platform": "apprise", "config": "/path/"}}
+
+    with patch("apprise.AppriseConfig.add", return_value=False):
+        assert await async_setup_component(hass, BASE_COMPONENT, config)
+        await hass.async_block_till_done()
+
+    with patch("apprise.AppriseConfig.add", return_value=True):
+        with patch("apprise.Apprise.add", return_value=False):
+            assert await async_setup_component(hass, BASE_COMPONENT, config)
+            await hass.async_block_till_done()
+
+
+async def test_apprise_config_load_okay(hass, tmpdir):
+    """Test apprise configuration failures."""
+
+    # Test cases where our URL is invalid
+    t = tmpdir.mkdir("apprise-config").join("apprise")
+    t.write("mailto://user:pass@example.com/")
+
+    config = {"apprise": {"name": "test", "platform": "apprise", "config": str(t)}}
+
+    assert await async_setup_component(hass, BASE_COMPONENT, config)
+    await hass.async_block_till_done()
+
+
+async def test_apprise_url_load_fail(hass):
+    """Test apprise url failure."""
+
+    config = {
+        "apprise": {
+            "name": "test",
+            "platform": "apprise",
+            "url": "mailto://user:pass@example.com",
+        }
+    }
+    with patch("apprise.Apprise.add", return_value=False):
```

Comment:
```suggestion
    with patch("homeassistant.components.apprise.notify.Apprise.add", return_value=False):
```

Reply:
It's working as is, but would you still prefer i change this (see comment below)

Reply:
Ok.